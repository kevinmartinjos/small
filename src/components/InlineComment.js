import React from 'react';

var PropTypes = React.PropTypes;

var InlineComment = React.createClass({
	PropTypes: {
		x: PropTypes.number.isRequired,
		y: PropTypes.number.isRequired
	},
	getInitialState(){
		return {
			style:{
				position: 'absolute',
				left: 0,
				top: 0,
				color: 'lightGreen'
			}
		};
	},
	componentDidMount(){
		this.setState({
			style: {
				position: 'absolute',
				left: this.props.x,
				top: this.props.y
			}
		});
	},
	render() {
		return(
			<div className="InlineComment" ref="child" style={this.state.style}>
				hello
			</div>
		);
	}
});

export default InlineComment;
