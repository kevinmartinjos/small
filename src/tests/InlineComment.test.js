import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import InlineComment from '../components/InlineComment';
import ReactTestUtils from 'react-addons-test-utils';

test('InlineComment should be rendered', () => {
	const component = renderer.create(
		<InlineComment />
	);
	let tree = component.toJSON();
  	expect(tree).toMatchSnapshot();
});
