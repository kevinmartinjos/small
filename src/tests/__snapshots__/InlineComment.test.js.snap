exports[`test InlineComment should be rendered 1`] = `
<div
  className="InlineComment" />
`;
